.github/workflows/build-containers.yml

```diff
@@ -1,60 +1,135 @@
 name: Build Containers
 
 on:
+  pull_request:
+    paths:
+      - .github/workflows/build-containers.yml
+      - devops/docker/**/Dockerfile
   push:
-    branches: [main, future/**]
-    paths: [devops/docker/**/Dockerfile]
+    branches:
+      - main
+    paths:
+      - .github/workflows/build-containers.yml
+      - devops/docker/**/Dockerfile
+
+env:
+  BUILD_TYPE: Release
+  MOUNT: /azure-osconfig
+  REGISTRY: ghcr.io
 
 jobs:
-  list-containers:
-    name: List modified containers
+  modified-containers:
+    name: Modified containers
     runs-on: ubuntu-latest
     outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      containers: ${{ steps.matrix.outputs.containers }}
     steps:
       - uses: actions/checkout@v2
 
-      - name: Get changed files
-        id: changed-files
+      - name: Get changed containers
+        id: changed-containers
         uses: tj-actions/changed-files@v18.7
         with:
           files: ./devops/docker/**/Dockerfile
 
-      - id: set-matrix
-        run: echo "matrix=$(echo -n ${{ steps.changed-files.outputs.all_changed_files }} | jq -R -s -c 'split(" ")')" >> $GITHUB_OUTPUT
+      - name: Create build matrix
+        id: matrix
+        run: |
+          # If this workflow is present in other_changed_files, then rebuild all containers
+          if [[ "${{ steps.changed-containers.outputs.other_changed_files }}" == *".github/workflows/build-containers.yml"* ]]; then
+            echo "Workflow file changed, adding all containers to build matrix..."
+            containers=$(ls -d devops/docker/**/Dockerfile | jq -R . | jq -s .)
+          else
+            containers=$(echo -n ${{ steps.changed-containers.outputs.all_changed_files }} | jq -R -s -c 'split(" ")')
+          fi
+
+          echo Containers to build: $containers
+          echo containers=$containers >> $GITHUB_OUTPUT
 
-  docker:
-    name: Build and push
-    needs: list-containers
-    runs-on: [self-hosted, 1ES.Pool=azure-osconfig-linux-pool]
```

Comment:
We must have the containers built on 1ES images, why revert to standard runners?

Reply:
During testing, the 1ES runners were consistently losing connection because the VMs in our subscription were being shutdown (this was preventing longer ARM builds from passing successfully). This could have been due to some cleanup/maintenance we were doing there last week. 

I will change it back to the 1ES runners in the meantime and monitor this to see if the issue persists.